### PR TITLE
add git submodule instructions to  contributing.md

### LINKED ISSUE #51355: Include instructions to cleanly remove third_party dep
## 📚 Documentation

after #50998 I couldn't checkout any commit that still uses `github.com/seemethere/pybind11.git` because the actual third_party commit lives on @seemethere 's personal fork.

the instruction on CONTRIBUTING.md to clean the repo doesn't work for this scenario when `.gitmodules` file was changed with a different repository.

cc @jlin27 @mruberry

### PR DESCRIPTION
**Pull request by walterddr**
Fixes #51355

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -150,7 +150,19 @@ with `brew install cmake` if you are developing on MacOS or Linux system.
   ```bash
   ENV_KEY1=ENV_VAL1[, ENV_KEY2=ENV_VAL2]* python setup.py develop
   ```
+* If you run into issue running `git submodule update --init --recursive`. Please try the following:
+  - If you encountered error such as 
+  ```
+  fatal: unable to access 'https://github.com/pybind11/pybind11.git': could not load PEM client certificate ...
+  ```
+  check whether your `git config --global --list` contains valid SSL certificates. If you are using http proxy, checkout for keyword `http.proxysslcert` as well.
```
Comment (janeyx99) [inline CONTRIBUTING.md:158]: What are valid SSL certs supposed to look like?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -150,7 +150,19 @@ with `brew install cmake` if you are developing on MacOS or Linux system.
...
 
+  - If you encountered error such as
+  ```
+  error: Submodule 'third_party/pybind11' could not be updated
+  ```
+  check whether your `git config --global --list` contains any `submodule.*` configs. Remove them in your `~/.gitconfig` and try again.
+  -
```
Comment (janeyx99) [inline CONTRIBUTING.md:165]: is this last - supposed to be here?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -150,7 +150,19 @@ with `brew install cmake` if you are developing on MacOS or Linux system.
...
+  ```
+  fatal: unable to access 'https://github.com/pybind11/pybind11.git': could not load PEM client certificate ...
+  ```
+  check whether your `git config --global --list` contains valid SSL certificates. If you are using http proxy, checkout for keyword `http.proxysslcert` as well.
 
+  - If you encountered error such as
+  ```
+  error: Submodule 'third_party/pybind11' could not be updated
+  ```
+  check whether your `git config --global --list` contains any `submodule.*` configs. Remove them in your `~/.gitconfig` and try again.
```
Comment (janeyx99) [inline CONTRIBUTING.md:164]:
```suggestion
  check whether your `git config --global --list` contains any `submodule.*` configs. Remove them from your `~/.gitconfig` and try again.
```
nit (also is everyone's git config stuff in ~/.gitconfig?)

File: CONTRIBUTING.md (hunk 2 above)
Comment (walterddr) [reply CONTRIBUTING.md:165]: yeah. let me rephrase it

File: CONTRIBUTING.md (hunk 1 above)
Comment (walterddr) [reply CONTRIBUTING.md:158]: (hmm. let me try to come up with a good example, but great feedback)

File: CONTRIBUTING.md (hunk 3 above)
Comment (walterddr) [reply CONTRIBUTING.md:164]:
```suggestion
  check whether your `git config --global --list` contains any `submodule.*` configs. Remove them from your global config, (usually in `$HOME/.gitconfig`) and try again.
```
